snowpack/src/hmr-server-engine.ts

```diff
@@ -110,9 +118,33 @@ export class EsmHmrEngine {
   }
 
   broadcastMessage(data: object) {
+    if (this.liveReloadDelayMs > 0) {
+      if (this.currentBatchTimeout) {
+        clearTimeout(this.currentBatchTimeout);
+      }
+      this.currentBatch.push(data);
+      this.currentBatchTimeout = setTimeout(() => this.broadcastBatch(), this.liveReloadDelayMs);
+    } else {
+      this.internalBroadcastMessage([data]);
+    }
+  }
+
+  broadcastBatch() {
+    if (this.currentBatchTimeout) {
+      clearTimeout(this.currentBatchTimeout);
+    }
+    if (this.currentBatch.length > 0) {
+      this.internalBroadcastMessage(this.currentBatch);
+      this.currentBatch = [];
+    }
+  }
+
+  private internalBroadcastMessage(dataArr: object[]) {
     this.clients.forEach((client) => {
       if (client.readyState === WebSocket.OPEN) {
-        client.send(JSON.stringify(data));
+        dataArr.forEach((data) => {
```

Comment:
One thing that's missing here: if there's a single `reload` event inside of `dataArr`, we should replace the entire batch with just one `reload` event.  This is because a `reload` event triggers the entire browser session to reload, rendering any other specific HMR events invalid/useless.